doc/source/tutorials/add_model.rst

```diff
@@ -0,0 +1,327 @@
+
+1. **Navigate to the model family's asset card file:**
+
+    .. code-block:: bash
+
+        src/fairseq2/assets/cards/models/qwen.yaml
+
+2. **Add a new asset card entry:**
+
+    .. code-block:: yaml
+
+        name: qwen25_3b_instruct
+        model_family: qwen
+        model_arch: qwen25_3b
+        checkpoint: "hg://qwen/qwen2.5-3b-instruct"
+        tokenizer: "hg://qwen/qwen2.5-3b-instruct"
+        tokenizer_family: qwen
+        tokenizer_config:
+            use_im_end: true
+
+- ``name``: The model name users will use (e.g., ``load_model("qwen25_3b_instruct")``)
+- ``model_family``: Which model family handles this model (``qwen``)
+- ``model_arch``: Which architecture configuration to use (``qwen25_3b``)
+- ``checkpoint``: Where to download the model weights from
+- ``tokenizer``: Where to download the tokenizer from
+- ``tokenizer_family``: Which tokenizer family to use
+- ``tokenizer_config``: Tokenizer-specific settings
+
+For more details on asset card options, see :ref:`basics-assets`.
+
+Step 3: Verify the Integration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+After adding the configuration and asset card, verify that your model is properly registered:
+
+1. **Check if model is recognized:**
+
+   .. code-block:: bash
+
+       # List all models to see if yours appears
+       python -m fairseq2.assets list --kind model
+
+       # Look specifically for your model
+       python -m fairseq2.assets list --kind model | grep qwen25_3b_instruct
+
+2. **Test model loading:**
+
+   .. code-block:: python
+
+       import fairseq2
+       from fairseq2.models.hub import load_model
+
+       # Test loading your model
+       try:
+           model = load_model("qwen25_3b_instruct")
+           print(f"✓ Success! Loaded model with {sum(p.numel() for p in model.parameters())} parameters")
+       except Exception as e:
+           print(f"✗ Error: {e}")
+
+3. **Inspect model metadata:**
+
+   .. code-block:: bash
+
+       # Show detailed model information
+       python -m fairseq2.assets show qwen25_3b_instruct
+
+Asset Source Options
+--------------------
+
+fairseq2 supports multiple sources for model checkpoints and tokenizers:
+
+Hugging Face Hub (Recommended)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Most common and convenient option:
+
+.. code-block:: yaml
+
+    checkpoint: "hg://qwen/qwen2.5-3b-instruct"
+    tokenizer: "hg://qwen/qwen2.5-3b-instruct"
+
+Note that only safetensors are supported for checkpoints.
+
+Local Files
+^^^^^^^^^^^
+
+For development or custom models:
+
+.. code-block:: yaml
+
+    checkpoint: "file:///path/to/my/model.pt"
+    tokenizer: "file:///path/to/my/tokenizer"
+
+HTTP URLs
+^^^^^^^^^
+
+Direct download links:
+
+.. code-block:: yaml
+
+    checkpoint: "https://example.com/models/my_model.pt"
+
+
+Common Model Parameters
+-----------------------
+
+When creating new architecture configurations, here are the most common parameters you'll need to set:
```

Comment:
We should mention that these parameter names are "conventions" used by most of our models, but not necessarily standard to all models (e.g. wav2vec2 does not have a `vocab_size` or NLLB does not have `num_key_value_heads`).